File: tests/bindings/delphi/TestBrickletIndustrialDualRelay.pas
program TestBrickletIndustrialDualRelay;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletIndustrialDualRelay;

type
  TTBIDR = class
  private
    ipcon: TIPConnection;
    dr: TBrickletIndustrialDualRelay;
  public
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = 'xyz'; { Change to your UID }

var
  e: TTBIDR;

procedure TTBIDR.Execute;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  dr := TBrickletIndustrialDualRelay.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Turn both relays off and on }
  dr.SetValue(false, false);
  Sleep(1000);
  dr.SetValue(true, true);

  ipcon.Destroy; { Calls ipcon.Disconnect internally }
end;

begin
  e := TTBIDR.Create;
  e.Execute;
  e.Destroy;
end.

File: tests/bindings/javascript/node.js/TestBrickletIndustrialDualRelay.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'xyz'; // Change to your UID

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var dr = new Tinkerforge.BrickletIndustrialDualRelay(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        // Turn both relays off and on
        dr.setValue(false, false);

        setTimeout(function() {
            dr.setValue(true, true);

            setTimeout(function() {
                process.exit(0);
            }, 500);
        }, 1000);
    }
);
